### ISSUE TITLE
Support arbitrary metrics

### ISSUE THREAD

**Issue by MasonProtter** [chakravala/Grassmann.jl#8]
It might be nice to support fully general, non-diagonal metrics at some point.

**Comment by chakravala** [comment]
Support for arbitrary metrics is also going to require the `VectorSpace` definitions to be updated in

DirectSum.jl : https://github.com/chakravala/DirectSum.jl

Have you familiarized yourself with that package yet?

**Comment by rleegates** [comment]
This is a great package! Unfortunately, this issue an the GPL are keeping me from using it.

**Comment by chakravala** [comment]
This issue will be addressed in either v0.3 or v0.4 of *DirectSum.jl* (after the extension I am currently designing for tensor fields). For me, it doesn't make sense to address this issue until the tensor fields are fully implemented.

The license of the repository is non-negotiable; however, it may be possible to separately negotiate individual licenses for specific users. This repository is the result of my own personal blood, sweat, and tears. If you want to modify and redistribute my software, GPL must be used or a very big price must be paid.

**Comment by rleegates** [comment]
That’s great news! I’ll check it out when it’s available. 

I am sure this was a lot of work, I implemented the (bundle-valued) multivector/differential form algebra, calculus on manifolds, in closed source, and it took quite some iterations to get it to run efficiently enough for my needs. So I do understand your position completely, this appears to be much more general and more cleanly implemented. So again, thanks for your hard work.

There have been cases though where a quick note has been enough for the repository owner to say: “Oops, MIT’s fine, too!” so I thought I’d raise the issue.

**Comment by chakravala** [comment]
Some of my repositories have the MIT license, but not this repository. MIT license would be negative for this project and also disrespectful to my expended effort on the whole foundation. For this kind of software, I would have either chosen closed source or GPL, but not MIT license. This is a special and unique effort, and making it closed source would not have benefited the project, so GPL it is.

Sprechen sie Deutsch? I am also german

**Comment by rleegates** [comment]
Ja, ich spreche deutsch! Wenn sie mögen, schreiben Sie mir doch eine persönliche Nachricht.

My implementation is written in Julia v0.7.

**Comment by chakravala** [comment]
this issue has been marked as `pay-donate` because I don't currently intend to implement this feature unless I am paid to work on it, since I am otherwise pre-occupied @rleegates @MasonProtter
